docker-compose/docker-compose.yaml

```diff
@@ -160,7 +160,7 @@ services:
   gitserver-0:
     container_name: gitserver-0
     image: 'index.docker.io/sourcegraph/gitserver:insiders@sha256:4f67f43a5bbefef5829da08be8f96235935d048218a22b96dd9f8af6f7fe7256'
-    cpus: 4
+    cpu_shares: 4096
     mem_limit: '8g'
     environment:
       - GOMAXPROCS=4
```

Comment:
Do we need to review the GOMAXPROCS settings, or does everything just work out like before? Should we remove these env var's entirely?

Reply:
I suggest we remove these entirely
cc @sourcegraph/repo-management

Reply:
To be extra clear (since repo-management is tagged here): we set GOMAXPROCS on all of the Sourcegraph-proper services (frontend, gitserver, etc.), and we're proposing that we remove that setting from all the services.

Reply:
Nothing like tripping over a fundamental language Golang issue in a Github comment: https://github.com/golang/go/issues/33803 
But for this case, yes, we want to drop these everywhere. If the CPU isn't being actively used by a higher priority process it should be allowed to use all cores.

Reply:
An issue opened almost 3 years ago in golang... woof!

Reply:
https://github.com/uber-go/automaxprocs might work here. 
Before merging we should test this on devmanaged

Reply:
Deployed to devmanaged, and it looks like this does affect some cadvisor CPU metrics. Most of the charts look ok but this one is [missing CPU data since the update](https://devmanaged.sourcegraph.com/-/debug/grafana/explore?orgId=1&left=%5B%221644239097280%22,%221644260697280%22,%22Prometheus%22,%7B%22expr%22:%22cadvisor_container_cpu_usage_percentage_total%7Bname%3D~%5C%22%5Erepo-updater.*%5C%22%7D%22,%22interval%22:%22100ms%22,%22datasource%22:%22Prometheus%22,%22exemplar%22:true,%22requestId%22:%22Q-5613d068-0719-4e42-a073-852f2e95c8cc-0A%22%7D%5D)